### PR TITLE
Adds check-rabbitmq-queues-synchronised.rb

### PR DESCRIPTION
**Pull request by cyrilgdn**
#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [x] Tests

- [x] Add the plugin to the README

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose

This check will verify that all mirrored queues are synchronised.

#### Examples
```

$ bin/check-rabbitmq-queues-synchronised.rb -h rabbitmq.service.consul --username guest --password guest              
CheckRabbitMQQueuesSynchronised OK

# With an unsynchronised queue
$ bin/check-rabbitmq-queues-synchronised.rb -h rabbitmq.service.consul --username guest --password guest
CheckRabbitMQQueuesSynchronised CRITICAL: 1 unsynchronised queues

$ bin/check-rabbitmq-queues-synchronised.rb -h rabbitmq.service.consul --username guest --password guest --list-queues
CheckRabbitMQQueuesSynchronised CRITICAL: test_queue: 1 unsynchronised slave(s)
```

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

**Comment by cyrilgdn** [comment]
@majormoses I added unit tests for this check (and by the way learned how to make tests in ruby :))
and some examples in this PR.

**Comment by majormoses** [comment]
I will get to this hopefully tonight.

File: bin/check-rabbitmq-queues-synchronised.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  def run
+    @crit = []
+
+    queues = get_queues config
+
+    queues.each do |q|
+      next unless q.key?('slave_nodes')
+
+      nb_slaves = q['slave_nodes'].count
+      next unless nb_slaves != 0
```
Comment (majormoses) [inline bin/check-rabbitmq-queues-synchronised.rb]: `next if nb_slaves.zero?` I think is more readable

File: bin/check-rabbitmq-queues-synchronised.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+# main plugin class
+class CheckRabbitMQQueuesSynchronised < Sensu::Plugin::RabbitMQ::Check
+  option :list_queues,
+         description: 'If set, will ouput the list of all unsynchronised queues, otherwise only the count',
+         long: '--list-queues',
+         boolean: true,
+         default: false
+
+  def run
+    @crit = []
```
Comment (majormoses) [inline bin/check-rabbitmq-queues-synchronised.rb]: if it's a list it should be plural to indicate that it is not a singular item.

File: bin/check-rabbitmq-queues-synchronised.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    queues = get_queues config
+
+    queues.each do |q|
+      next unless q.key?('slave_nodes')
+
+      nb_slaves = q['slave_nodes'].count
+      next unless nb_slaves != 0
+      unsynchronised = nb_slaves - q['synchronised_slave_nodes'].count
+      if unsynchronised != 0
+        @crit << "#{q['name']}: #{unsynchronised} unsynchronised slave(s)"
```
Comment (majormoses) [inline bin/check-rabbitmq-queues-synchronised.rb]: `@crits`

File: bin/check-rabbitmq-queues-synchronised.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      end
+    end
+    if @crit.empty?
```
Comment (majormoses) [inline bin/check-rabbitmq-queues-synchronised.rb]: `@crits`

File: bin/check-rabbitmq-queues-synchronised.rb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      ok
+    elsif config[:list_queues]
+      critical @crit.join(' - ')
```
Comment (majormoses) [inline bin/check-rabbitmq-queues-synchronised.rb]: `@crits.join(' - ')`

**Comment by cyrilgdn** [comment]
@majormoses Updated with your comments taken in account.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-rabbitmq/versions/3.6.0

**Pull request merged**